jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java

```diff
@@ -14,112 +14,120 @@
+        assertThat(map).isNotNull().isEmpty();
+      }
+
+      @Test
+      @DisplayName("from non-existing file, config should be empty")
+      void configFileDoesNotExist_shouldBeEmpty() {
+        final String nonExistingFile = "/bla/blub/lalala/config.yml";
+        System.setProperty("vertx.config.test", nonExistingFile);
+        try {
+          Map<String, Integer> map = extractFromFile("vertx.config.test", null);
+          assertThat(map).isNotNull().isEmpty();
+        } finally {
+          System.getProperties().remove("vertx.config.test");
         }
+      }
     }
 
-    @Test
-    public void testKeyPatterns() {
-        Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-keys.yml");
-
-        Object[] testData = {
-                "web.port", true,
-                "web_port", true,
-                "webPort", true,
-                "ssl.support", false,
-                "ports", false,
-                "ports.http", false,
-                "ports.https", false
-        };
-
-        for (int i = 0; i < testData.length; i +=2 ) {
-            assertEquals(testData[i+1], map.containsKey(testData[i]));
-        }
+    @ParameterizedTest(name = "''{0}'' should be ''{1}''")
+    @MethodSource("patterns")
+    void keyPatterns(String key, boolean expected) {
+      Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-keys.yml");
+      assertThat(map.containsKey(key)).isEqualTo(expected);
     }
 
-    @Test
-    public void testAddPortToList() {
-        Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-values.yml");
-
-        Object[] testData = {
-                "http.port", 8080,
-                "https.port", 443,
-                "ssh.port", 22,
-                "ssl.enabled", null
-        };
-        for (int i = 0; i < testData.length; i +=2 ) {
-            assertEquals(testData[i+1], map.get(testData[i]));
-        }
+    static Stream<Arguments> patterns() {
+      return Stream.of(
+          arguments("web.port", true),
+          arguments("web_port", true),
+          arguments("webPort", true),
+          arguments("ssl.support", false),
+          arguments("ports", false),
+          arguments("ports.http", false),
+          arguments("ports.https", false));
     }
 
-    @Test
-    public void testNoProperty() {
-        Map<String, Integer> map = extractFromFile(null, getClass().getSimpleName() + ".yml");
-        assertNotNull(map);
-        assertEquals(0,map.size());
+    @DisplayName("add port to list")
+    @ParameterizedTest(name = "''{0}'' with port ''{1}''")
+    @MethodSource("ports")
+    void addPortToList(String portName, int expectedPort) {
+      Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-values.yml");
+      assertThat(map).containsEntry(portName, expectedPort);
```

Comment:
Why have you made these tests parameterized? I see only a constant input provided to test.

Reply:
I'm really sorry, it was due to some misunderstandings related to testData.
Fixed.